blocks/hooks/custom-anchor-styles.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getBlockRandomAnchor, hasBlockSupport, getBlockType } from '../api';
+
+const CUSTOM_STYLE_ATTRIBUTES = [ 'backgroundColor', 'textColor' ];
+
+class AddAnchorWhenNeeded extends Component {
+	componentWillReceiveProps( nextProps ) {
```

Comment:
Are we sure this is only necessary to check when props change, not also on the initial mount? Thinking a block which is prepopulated with custom styling attributes.

Reply:
If a block is prepopulated with custom styles my expectation is that the anchor is already present so we don't need to add it.
Regarding posts created before this change that's problematic. This change changes the markup e.g: changes inline style attributes so, right now we display the message saying the block is invalid. I have not yet addressed the migration, as soon as https://github.com/WordPress/gutenberg/pull/3673 or a similar approach is landed I will use the approach to migrate blocks